### PR TITLE
Update README to use continuous deployment.

### PR DESCRIPTION
**Pull request by squaresurf**
Why:

* Delivers value to the end user quicker.
* Simplifies the deployment pipeline by freeing up engineers to provide
more value via getting more work done.
* Ensures that code is done done before merging into master.

This change addresses the need by:

* Add deployment workflow to the README.

### CODE REVIEW

**Comment by squaresurf** [comment]
@samcdavid I need your :+1: before merging this.

**Comment by gabeweaver** [comment]
@squaresurf this is awesome! I do have a few questions / suggestions / feedback...

1. It might be useful to add specific document to how continuous deployments are actually working - like technically. 
2. I noticed on [this issue from ASME](https://github.com/spartansystems/asme/pull/224), you referenced a buildpack. Which buildpack are we using for CD?
3. Maybe I'm missing something, but is there a reason we aren't leveraging the [app.json manifest](https://blog.heroku.com/archives/2014/5/22/introducing_the_app_json_application_manifest) for storing configuration stuff as code and in version control so it's easy for others to collaborate and maintain? I'm sure you have a good reason...just trying to understand. 
4. There is also this sweet [apps-setup Heroku platform API endpoint](https://devcenter.heroku.com/articles/platform-api-reference#app-setup)...not sure if it would be useful for this...just making sure we looked at various options :) 

This is from the recent state of devops report and I couldn't agree more:
```
We found that the practices that make up continuous delivery — deployment automation and automated
testing, continuous integration, and version control for all production artifacts — have a significant predictive
relationship to deployment pain, IT performance and change fail rate. In turn, IT performance predicts
organizational performance, as measured by productivity, market share and profitability.3
```

Here is another [meaningful piece](https://www.dropbox.com/s/hlz0vjzt5ph6xq3/Screenshot%202016-02-02%2020.27.22.png?dl=0) of the report.

Long and short, this is an absolutely wonderful start and definitely headed in the right direction...i just want to encourage us to realize how much we can do with Heroku now and that we actually can have an enormous amount of control regarding our infrastructure if we want it. At the end of the day, version controlling any production artifact is probably the single most important thing we can focus on accomplishing now. GOOD WORK!

**Comment by gabeweaver** [comment]
@squaresurf I also think it'd be worth putting together a basic roadmap for the ideal CD setup. Rereading my comment above and the updated Readme made me realize we need to get away from the "+1" reaction in Slack channels ASAP. It's not really continuous deployment if we have to manually QA every single Deployment....it's a step closer...but it feels like we should have enough tooling setup that prevents broken things from going up, then detecting broken things and alerting us ASAP once an issue does make it to production. Instead of humans being the safety nets, we need to let machines be safety nets...and us hold the supervisory functions. Take a second to [read about autonomation](https://en.wikipedia.org/wiki/Autonomation).

Do you have any ideas for automated acceptance tests? I also think something as simple as rollbar exceptions going to the same slack channel as deploy notifications...and also setting up basic New Relic alerting & Key Transaction tracking...with notifications to the same Slack channel...would be a simple starting point we could do as early as tomorrow. If we don't have the proper feedback loops from our tools...in real time....once we deploy....we are sitting ducks for something catastrophic IMHO...just posting notes/thoughts/random stream here before I forgot them during my sleep tonight.

File: README.md [hunk 1]
```diff
@@ -67,8 +67,25 @@ Note: add stuff for rspec bin stub.
...
+* We review all our code on Github before merging into master and when we do we keep in mind that the code will go to production once merged.
+* All deployments post a notification in the project slack channel so that there is clear communication when deployments happen. If the notification is for a production deploy, then whoever merged into master is resposible for checking production and posting a +1 in the slack channel or fixing any issues that the deploy may have caused (fixing may include asking for help).
+* Deploys only happen if our tests pass for the code being deployed.
+
+### Database Migrations
+
+* Are reviewed during the code review process.
+* Database backups can be manually created via heroku and should be enabled to run automatically.
+* When the code is merged into master a database migration should be performed as part of the app build process in order to avoid downtime for the deploy.
+* If there is an issue with a database migraiton the rollback plan (whether implemented in code or a manual process) should be executed. The individual who initiated the deploy will be responsible for understanding the rollback plan and ensuring its execution.
```
Comment (samcdavid) [inline README.md]: Spelling s/migraiton/migration

**Comment by squaresurf** [comment]
@gabeweaver thank you so much for the meaningful review!!!

I'm gonna try to address everything, let me know if I miss anything.

1. I'm not sure what to add technically other than the buildpack bit. It really isn't very technical as it uses heroku's deploy mechanism. I guess I could lay all that out if you think that would be good.
2. Good catch. I'll add that :smile:
3. I spent a few hours a couple weeks ago researching the app.json to make it so we could add infra layouts to our repos, but found it wasn't as simple as I was hoping to spin a new app from the json. I'll research it a little more. I might just write a script that sets it all up via the heroku cli tool. In the end, this is very important to get done, but I don't think it should block this process moving forward.
4. I think that api is what is needed to create an app from the app.json, but I didn't see a simple way to use it in our scenario from my reading. I should probably just do a reasearch spike where I try it out quickly.

As for having infra code in our repos, I couldn't agree more. In fact it was one of the things I highlighted as I led the agile book club last week on releasing. I think it is something we must do in order to bring consistency to our infra across apps. More importantly in my opinion is a script to check conformance with our best practices so we can quickly see where the gaps are in an app's infra.

The +1 manual qa is a stop gap to get us to our end goal. I want it to be painful so that we start using our monitoring, and testing services fully so we don't have rely on faulty human tests. This is the first place of improvement I am planning on focusing on. Autonomation is awesome. In fact, that is one of the things I enjoyed most about using chef and sensu in the past.

I will inspect the tool integration improvements you made by the end of this week:

> I also think something as simple as rollbar exceptions going to the same slack channel as deploy notifications...and also setting up basic New Relic alerting & Key Transaction tracking...with notifications to the same Slack channel...would be a simple starting point we could do as early as tomorrow.

As for this PR, I don't think any of this blocks this PR from going through as these **real** problems are in existence whether we adopt this process or not. Please let me know if you feel differently.

**Comment by samcdavid** [comment]
I'm good with these changes, as long as everyone else likes them, I'm in :+1:

**Comment by squaresurf** [comment]
I've updated the README to point at our wiki. Please see the PR for that addition here: https://github.com/spartansystems/spartan/pull/43

**Pull request merged**